Q10_JJ.Rmd

```diff
@@ -0,0 +1,131 @@
+---
+title: "Untitled"
+author: "Jinshi"
+date: "7/20/2020"
+output: html_document
+---
+
+
+## Jinshi: take one dataset as an example
+```{r}
+first_rh_dset <- csr_dataset("d20190424_ZHANG_maple")
+first_rh_data <- first_rh_dset$data
+colnames(first_rh_data)
+# There are 7 columns, and each column including:
+# "CSR_PORT": port number
+# "CSR_TIMESTAMP_BEGIN": Begin time
+# "CSR_TIMESTAMP_END": end time
+# "CSR_FLUX": Rh flux
+# "CSR_R2": linear fitting R2
+# "CSR_SM30": Soil moisture           
+# "CSR_T5": Soil temperature
+
+# Check NA flux and T5
+# You can see that there are no missing Rh, but 43 T5 missing values
+first_rh_data %>% 
+  select(CSR_FLUX, CSR_T5) %>% 
+  mutate(
+    RH_YN = case_when(
+      !is.na(CSR_FLUX)  ~ "YES",
+      TRUE ~ "NO"),
+    T5_YN = case_when(
+      !is.na(CSR_T5) ~ "YES",
+      TRUE ~ "NO")) %>% 
+  count(RH_YN, T5_YN)
+
+# because colname may differ in different dataset, it is better we rename the flux and T5 column
+# we also need to calculate the log(Rh) and we need to remove negative fluxes before log transform
+first_rh_data %>% 
+  rename('CSR_Rh' = 'CSR_FLUX',
+         'CSR_TS' = 'CSR_T5') %>% 
```

Comment:
I think for Colin's work at this point we'll want to add a depth field, e.g.
```
mutate(Depth = 5) %>%
```

Reply:
Yes, that's a good suggestion